source/tests/system/nirfmxwlan_driver_api_tests.cpp

```diff
@@ -339,6 +383,794 @@ TEST_F(NiRFmxWLANDriverApiTests, TXPFromExample_FetchData_DataLooksReasonable)
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_composite_rmsevm_response.composite_data_rms_evm_mean());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_composite_rmsevm_response.composite_pilot_rms_evm_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_numberof_symbols_used_response);
+  EXPECT_EQ(16, ofdm_mod_acc_fetch_numberof_symbols_used_response.number_of_symbols_used());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_frequency_error_mean_response);
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_frequency_error_mean_response.frequency_error_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_symbol_clock_error_mean_response);
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_symbol_clock_error_mean_response.symbol_clock_error_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_iq_impairments_response);
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.relative_iq_origin_offset_mean());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_gain_imbalance_mean());
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_quadrature_error_mean());
+  EXPECT_GT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.absolute_iq_origin_offset_mean());
+  EXPECT_LT(0.0, ofdm_mod_acc_fetch_iq_impairments_response.iq_timing_skew_mean());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_ppdu_type_response);
+  EXPECT_EQ(0, ofdm_mod_acc_fetch_ppdu_type_response.ppdu_type());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_mcs_index_response);
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_guard_interval_type_response);
+  EXPECT_EQ(0, ofdm_mod_acc_fetch_guard_interval_type_response.guard_interval_type());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_lsig_parity_check_status_response);
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_sigcrc_status_response);
+  EXPECT_EQ(-1, ofdm_mod_acc_fetch_sigcrc_status_response.sig_crc_status());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_sigbcrc_status_response);
+  EXPECT_EQ(-1, ofdm_mod_acc_fetch_sigbcrc_status_response.sigb_crc_status());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_pilot_constellation_trace_response);
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_pilot_constellation_trace_response.pilot_constellation(0).real());
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_pilot_constellation_trace_response.pilot_constellation(0).imaginary());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_data_constellation_trace_response);
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_data_constellation_trace_response.data_constellation(0).real());
+  EXPECT_NE(0.0, ofdm_mod_acc_fetch_data_constellation_trace_response.data_constellation(0).imaginary());
+  EXPECT_SUCCESS(session, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response);
+  EXPECT_EQ(-26, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.x0());
+  EXPECT_EQ(1, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.dx());
+  EXPECT_EQ(53, ofdm_mod_acc_fetch_chain_rmsevm_per_subcarrier_mean_trace_response.chain_rms_evm_per_subcarrier_mean_size());
```

Comment:
I am sure this is just from generation but is there really much value for checking both the size of the array and the size we pass in ? We have unit tests for this

Reply:
Yeah, this is from the generation. We could possibly trim that down on the generation side with David's script but I'm going to leave them as is for now. Not worth the manual cleanup to remove redundant validation.